spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -0,0 +1,186 @@
+ */
+public class SparkSessionCatalog<T extends TableCatalog & SupportsNamespaces>
+    implements TableCatalog, SupportsNamespaces, CatalogExtension {
+
+  private String name = null;
+  private SparkCatalog icebergCatalog = null;
+  private T sessionCatalog = null;
+  private boolean createParquetAsIceberg = false;
+  private boolean createAvroAsIceberg = false;
+
+  /**
+   * Build a {@link SparkCatalog} to be used for Iceberg operations.
+   * <p>
+   * The default implementation creates a new SparkCatalog with the session catalog's name and options.
+   *
+   * @param catalogName catalog name
+   * @param options catalog options
+   * @return a SparkCatalog to be used for Iceberg tables
+   */
+  protected SparkCatalog buildSparkCatalog(String catalogName, CaseInsensitiveStringMap options) {
+    SparkCatalog newCatalog = new SparkCatalog();
+    newCatalog.initialize(catalogName, options);
+    return newCatalog;
+  }
+
+  @Override
+  public String[][] listNamespaces() throws NoSuchNamespaceException {
+    return sessionCatalog.listNamespaces();
+  }
+
+  @Override
+  public String[][] listNamespaces(String[] namespace) throws NoSuchNamespaceException {
+    return sessionCatalog.listNamespaces(namespace);
+  }
+
+  @Override
+  public Map<String, String> loadNamespaceMetadata(String[] namespace) throws NoSuchNamespaceException {
+    return sessionCatalog.loadNamespaceMetadata(namespace);
+  }
+
+  @Override
+  public void createNamespace(String[] namespace, Map<String, String> metadata) throws NamespaceAlreadyExistsException {
+    sessionCatalog.createNamespace(namespace, metadata);
+  }
+
+  @Override
+  public void alterNamespace(String[] namespace, NamespaceChange... changes) throws NoSuchNamespaceException {
+    sessionCatalog.alterNamespace(namespace, changes);
+  }
+
+  @Override
+  public boolean dropNamespace(String[] namespace) throws NoSuchNamespaceException {
+    return sessionCatalog.dropNamespace(namespace);
+  }
+
+  @Override
+  public Identifier[] listTables(String[] namespace) throws NoSuchNamespaceException {
+    // delegate to the session catalog because all tables share the same namespace
+    return sessionCatalog.listTables(namespace);
+  }
+
+  @Override
+  public Table loadTable(Identifier ident) throws NoSuchTableException {
+    try {
+      return icebergCatalog.loadTable(ident);
+    } catch (NoSuchTableException e) {
+      return sessionCatalog.loadTable(ident);
+    }
+  }
+
+  @Override
+  public Table createTable(Identifier ident, StructType schema, Transform[] partitions,
+                           Map<String, String> properties)
+      throws TableAlreadyExistsException, NoSuchNamespaceException {
+    String provider = properties.get("provider");
+    if (provider == null || "iceberg".equalsIgnoreCase(provider)) {
+      return icebergCatalog.createTable(ident, schema, partitions, properties);
+
+    } else if (createParquetAsIceberg && "parquet".equalsIgnoreCase(provider)) {
```

Comment:
So, this will create an Iceberg table when the provider is "parquet" and have an option named "parquet-enabled"?

Reply:
Yes. `USING parquet` would be interpreted as an Iceberg table create if you're using an Iceberg session catalog and have the property enabled.

Reply:
Yes. USING parquet would be interpreted as an Iceberg table create if you're using an Iceberg session catalog and have the property enabled.